bigquery/indexer.py

```diff
@@ -117,31 +97,14 @@ def _rows_from_export(
         blob.delete()
 
 
-# Sample and participant tables need to be indexed differently.
-# For participant tables, we can use partial updates
-# (https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#_updates_with_a_partial_document)
-#
-# If one participant table has weight and another has height:
-# - First weight table is indexed. Participant documents get a weight field.
-# - Then height table is indexed. Participant documents get a height field. The
-#   weight fields are unchanged.
-#
-# Now say one sample table has center and another has platform.
-# - First center table is indexed. For each participant document, a samples
-#   field is created. The samples field contains an array of nested objects,
-#   each of which has a center field.
-# - Then platform table is indexed. For each participant document, the samples
-#   field is overwritten to the new value, which contains platform and not
-#   center.
-# In order to keep the center field, one must use a script. See
-# https://discuss.elastic.co/t/updating-nested-objects/87586/2 and
-# https://www.elastic.co/guide/en/elasticsearch/reference/6.4/docs-update.html
-def _sample_scripts_by_id_from_export(
+def _add_sample_table_to_participant_docs(
         storage_client, bucket_name, export_obj_prefix, table_name,
-        participant_id_column, sample_id_column, sample_file_columns):
+        participant_id_column, sample_id_column, sample_file_columns,
+        participant_docs):
     for row in _rows_from_export(storage_client, bucket_name,
                                  export_obj_prefix):
         participant_id = row[participant_id_column]
+        sample_id = row[sample_id_column]
         del row[participant_id_column]
         row = {
             '%s.%s' % (table_name, k) if k != sample_id_column else k: v
```

Comment:
Do you mind rewriting this to be more readable? It's not clear from reading it what this does. Let's write an explicit if instead of ternary operator.